File: react-native/ios/flashcardaiapp/BenchmarkLogger.m
//
//  BenchmarkLogger.m
//  flashcardaiapp
//
//  Created by Aman Nirala on 16/08/25.
//
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(BenchmarkLogger, NSObject)
RCT_EXTERN_METHOD(log:(NSString *)level message:(NSString *)message)
@end

File: react-native/ios/BenchmarkLogger.swift
//
//  BenchmarkLogger.swift
//  flashcardaiapp
//
//  Created by Aman Nirala on 16/08/25.
//
import Foundation

@objc(BenchmarkLogger)
class BenchmarkLogger: NSObject {
  @objc
  func log(_ level: NSString, message: NSString) {
    let tag = "Benchmark"
    switch level as String {
      case "error": NSLog("[\(tag)] ❌ %@", message)
      case "warn":  NSLog("[\(tag)] ⚠️ %@", message)
      default:      NSLog("[\(tag)] %@", message)
    }
  }
}
